### ISSUE TITLE
Don't succeed at using new DocumentStart()/DocumentEnd()

### ISSUE THREAD

**Issue by Tiraflo** [aaubry/YamlDotNet#285]
Hello everyone,

I'm trying to serialize dummy objects in order to discover library YamlDotNet. While defining my "converter" class, the following methods go right when using the emitter:
- New Scalar()
- New MappingStart()/MappingEnd()
- New SequenceStart()/SequenceEnd()

However, when I wish to start a new document by using New DocumentStart() and New DocumentClose(), I finally obtain an exception:

> 
> An unhandled exception of type 'YamlDotNet.Core.YamlException' occurred in YamlDotNet.dll
> 
> Additional information: Expected SCALAR, SEQUENCE-START, MAPPING-START, or ALIAS, got DocumentStart
> 

Here is a part of my sample where the problems occurs:

```
    Public Sub WriteYaml(emitter As IEmitter, value As Object, type As Type) Implements IYamlTypeConverter.WriteYaml
        Console.WriteLine("WriteYaml")
        emitter.Emit(New DocumentStart())
        'emitter.Emit(New DocumentStart(New Tokens.VersionDirective(New Version(3, 2)), New TagDirectiveCollection(), False))
        emitter.Emit(New Scalar("dummyAnchor", "dummyTag", "dummyValue", ScalarStyle.Plain, False, False))
        emitter.Emit(New YamlDotNet.Core.Events.Scalar("dummyScalar"))
        emitter.Emit(New DocumentEnd(True))
        'Throw New NotImplementedException()
    End Sub
```

According to the message received with the exception, a scalar is expected at least to make it work but it fails anyway.
Do you have any idea?

Also, how to add the famous "%YAML 1.1" at the beginning of the file? Shoud I make it manually?

Best regards,

Tiraflo

**Comment by aaubry** [comment]
You should not be emitting a new document in a converter. What are you trying to achieve?

**Comment by Tiraflo** [comment]
I'm trying to serialize a new file (from scratch), containing several documents in it

**Comment by aaubry** [comment]
If you need to emit multiple documents, you will need to serialize each document separately. There is no support to achieve that from a converter. Actually the serializer does not support that, but you can use the underlying `IValueSerializer` to achieve that:

```c#
public static void Main()
{
	var valueSerializer = new SerializerBuilder()
		.BuildValueSerializer();

	var emitter = new Emitter(Console.Out);
	emitter.Emit(new StreamStart());
		
	emitter.Emit(new DocumentStart());
	valueSerializer.SerializeValue(emitter, new FirstDocument(), typeof(FirstDocument));
	emitter.Emit(new DocumentEnd(true));
		
	emitter.Emit(new DocumentStart());
	valueSerializer.SerializeValue(emitter, new[] { 1, 2, 3 }, typeof(int[]));
	emitter.Emit(new DocumentEnd(true));
		
	emitter.Emit(new StreamEnd());
}
```
https://dotnetfiddle.net/O2laA7

**Comment by Tiraflo** [comment]
Hi,

Thank you for the answer. I figured out my way to use DocumentStart/DocumentEnd was not the right one then I tried something similar to your sample except that I was using

```
var serializer= new SerializerBuilder().Build();
serializer.Serialize(emitter, Me) // Me because this code was executed from a Class method
```

But in this case, I obtained the same kind of error (than in my first message). Finally, it works with your example (I don't know what's the different between both but it does not matter a lot...)

The remaining problem is I would like to obtain this kind of document:

--- !myTag &myAnchor

Unfortunately, DocumentStart() does not handle these parameters. However, SequenceStart() does but the anchor is prior to the tag, which doesn't suit my need.

Thank you,

Tiraflo
